### ISSUE TITLE
Show a message on `git commit` in the terminal when `git.terminalGitEditor` is true

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#152488]
Something like this:

> Info: Git has been automatically configured to use VS Code as the commit editor. To disable, switch git.terminalGitEditor to false.

If we can have escape sequences there we could style it like this to indicate the message is coming from VS Code:

![image](https://user-images.githubusercontent.com/2193314/174330420-f3ba4c7f-b587-4722-9d23-add2045ca3a3.png)

**Comment by jpcastberg** [comment]
Seems this setting somehow got turned on by default? It was a little confusing today when my message editor suddenly became vscode instead of vim. Feels like this should be opt-in rather than opt-out.

**Comment by Tyriar** [comment]
@jpcastberg we had that discussion internally as I was more on your side. While we want the terminal to behave like other terminals, we expect the experience to generally be better in the editor so with the message proposed I'm much more comfortable going out with this and collecting feedback, since it's very obvious why it happened and is easy to switch off. 

We're also exploring contextual actions in the terminal which could make it even easier to turn off.

**Comment by Tyriar** [comment]
Yes it's only in Insiders right now, I think the plan is to have it on by default in Stable in the upcoming version, but with the message. Thanks for the feedback

**Comment by lszomoru** [comment]
After further discussions within the team we have decided to change the **default** value of the `git.terminalGitEditor` setting to `false`. Out of the box, users will still be able to use the full editor to enter a commit message when using the “Commit” action in the source control viewlet, and they will be able to enable the terminal integration if they toggle the `git.terminalGitEditor` setting. This change will be included in Monday's Insiders release (2022-06-27).
